### PR TITLE
Improvement/jenkins 42104 repos list filterable

### PR DESCRIPTION
**Pull request by cliffmeyers**
# Description
- See [JENKINS-42104](https://issues.jenkins-ci.org/browse/JENKINS-42104).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by cliffmeyers** [comment]
This CI build will fail until I republish a new core-js that reflects the new JDL. Going to wait until I get a bee to do that.

**Comment by cliffmeyers** [comment]
filtering-repo-list

@i386 @michaelneale

**Comment by i386** [comment]
Looking great :)

@brody please review

**Comment by michaelneale** [comment]
Functionally looks great - LGTM from me. Need to see it pass tests and ATH of course - but GO GO GO

**Comment by brody** [comment]
Looks good to me. 👍

File: blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx [hunk 1]
```diff
@@ -15,6 +15,17 @@ export default class GithubRepositoryStep extends React.Component {
         this.props.flowManager.saveSingleRepo();
     }
 
+    _getLoadingMessage() {
+        const { repositoriesLoading } = this.props.flowManager;
+        const count = this.props.flowManager.repositories.length;
+
+        if (repositoriesLoading) {
+            return `Loading Repositories... ${count} so far.`;
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx]: NIT: i18n

File: blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx [hunk 2]
```diff
@@ -51,11 +67,12 @@ export default class GithubRepositoryStep extends React.Component {
 
                 { flowManager.selectableRepositories.length > 0 &&
                 <div className="container">
-                    <List
+                    <FilterableList
                       className="repo-list"
                       data={sortedRepos}
                       onItemSelect={(idx, repo) => this.selectRepository(repo)}
                       labelFunction={repo => repo.name}
+                      filterFunction={(text, repo) => repo.name.toLowerCase().indexOf(text) !== -1}
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx]: for completeness you could `text.toLowerCase()` also

File: blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx [hunk 3]
```diff
@@ -31,9 +42,14 @@ export default class GithubRepositoryStep extends React.Component {
         const orgName = flowManager.selectedOrganization.name;
         const existingPipelineCount = flowManager.existingPipelineCount;
         const sortedRepos = flowManager.selectableRepositories.slice().sort(this._sortRepos);
+        const loading = flowManager.repositoriesLoading;
 
         return (
-            <FlowStep {...this.props} className="github-repo-list-step" title={title} disabled={disabled}>
+            <FlowStep {...this.props} className="github-repo-list-step" title={title} loading={loading} disabled={disabled}>
+                <div className="loading-msg">
+                    { this._getLoadingMessage()}
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx]: guessing this is part of the design? seems like a weird way to show loading status of something that appears below it in the page... I'd just have an infinite scroll list or something, but that's me.

Comment (kzantow) [inline blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx]: Reading the ticket now, though seems fine.

File: blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx (hunk 1 above)
Comment (cliffmeyers) [inline blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx]: Yep, all I18N work will be done as part of this ticket:  https://issues.jenkins-ci.org/browse/JENKINS-41139

File: blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx (hunk 2 above)
Comment (cliffmeyers) [inline blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx]: Good call here. Fixed in commit a61ab56

**Comment by michaelneale** [comment]
:bee: and ATH happy
:bee:
:bee:
:bee:

File: blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx (hunk 1 above)
Comment (michaelneale) [inline blueocean-dashboard/src/main/js/creation/github/steps/GithubRepositoryStep.jsx]: @kzantow speak 'merican!

**Comment by cliffmeyers** [comment]
Passed ATH: https://ci.blueocean.io/blue/organizations/jenkins/ATH-Jenkinsfile/detail/master/1581/pipeline/18

**Comment by cliffmeyers** [comment]
ATH passed: https://ci.blueocean.io/blue/organizations/jenkins/ATH-Jenkinsfile/detail/master/1584/pipeline/18

**Pull request merged**